arithmetization/src/main/java/net/consensys/linea/zktracer/module/romlex/RomLex.java

```diff
@@ -83,19 +82,16 @@ public int getCodeFragmentIndexByMetadata(final ContractMetadata metadata) {
       throw new RuntimeException("Chunks have not been sorted yet");
     }
 
-    for (int i = 0; i < sortedOperations.size(); i++) {
-      final RomOperation c = sortedOperations.get(i);
-      if (c.metadata().equals(metadata)) {
-        return i + 1;
-      }
+    List<RomOperation> romOps = lookup.get(metadata);
+    if (romOps == null) {
+      throw new RuntimeException(
+          "RomChunk with:"
+              + String.format("\n\t\taddress = %s", metadata.address())
+              + String.format("\n\t\tdeployment number = %s", metadata.deploymentNumber())
+              + String.format("\n\t\tdeployment status = %s", metadata.underDeployment())
+              + "\n\tnot found");
     }
-
-    throw new RuntimeException(
-        "RomChunk with:"
-            + String.format("\n\t\taddress = %s", metadata.address())
-            + String.format("\n\t\tdeployment number = %s", metadata.deploymentNumber())
-            + String.format("\n\t\tdeployment status = %s", metadata.underDeployment())
-            + "\n\tnot found");
+    return romOps.size();
```

Comment:
CFI is not the size (as it's not a list) but the index of the given RomOperation in the sortedRomOperations